doc/examples/realtime_servo/src/pose_tracking_tutorial.cpp

```diff
@@ -0,0 +1,228 @@
+  target_pose.pose.orientation.y = rotation.y();
+  target_pose.pose.orientation.z = rotation.z();
+  target_pose.pose.position.x = position.x();
+  target_pose.pose.position.y = position.y();
+  target_pose.pose.position.z = position.z();
+
+  return target_pose;
+}
+
+int main(int argc, char* argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("servo_tutorial");
+
+  auto marker_publisher =
+      node->create_publisher<visualization_msgs::msg::MarkerArray>("/path_markers", rclcpp::SystemDefaultsQoS());
+  auto pose_publisher = node->create_publisher<geometry_msgs::msg::PoseStamped>("/servo_node/pose_target_cmds",
+                                                                                rclcpp::SystemDefaultsQoS());
+
+  auto switch_input_client = node->create_client<moveit_msgs::srv::ServoCommandType>("/servo_node/switch_command_type");
+
+  auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+  executor->add_node(node);
+
+  // Spin the node.
+  std::thread executor_thread([&]() { executor->spin(); });
+
+  visualization_msgs::msg::MarkerArray marray;
+  std::vector<Eigen::Vector3d> path = getPath();
+
+  for (size_t i = 0; i < path.size(); i++)
+  {
+    marray.markers.push_back(getMarker(i, path[i], "panda_link0"));
+  }
+  marker_publisher->publish(marray);
+
+  // Create the door.
+  Door door(node);
+  door.rotateDoor(M_PI / 2);
+
+  // Default end-effector pose of the Panda arm.
+  auto ee_pose = Eigen::Isometry3d::Identity();
+  ee_pose.rotate(Eigen::AngleAxisd(M_PI, Eigen::Vector3d::UnitX()));
+  ee_pose.translate(Eigen::Vector3d(0.3, 0.0, 0.4));
+
+  // Create the ROS message.
+  auto request = std::make_shared<moveit_msgs::srv::ServoCommandType::Request>();
+  request->command_type = moveit_msgs::srv::ServoCommandType::Request::POSE;
+  auto response = switch_input_client->async_send_request(request);
+  if (response.get()->success)
+  {
+    RCLCPP_INFO_STREAM(node->get_logger(), "Switched to input type: Pose");
+  }
+  else
+  {
+    RCLCPP_WARN_STREAM(node->get_logger(), "Could not switch input to: Pose");
+  }
+
+  // Follow the trajectory
+
+  const double publish_period = 0.15;
+  rclcpp::WallRate rate(1 / publish_period);
+
+  size_t i = path.size() - 1;
+  while (i > 0)
+  {
+    auto target_pose = getPose(path[i], Eigen::Quaterniond(ee_pose.rotation()));
+    target_pose.header.stamp = node->now();
+    pose_publisher->publish(target_pose);
+    rate.sleep();
+    i--;
+  }
+
+  double door_angle = M_PI / 2;
+  const double step = 0.01745329;  // 1 degree in radian
+  i = 0;
+  while (i < path.size())
+  {
+    ee_pose.rotate(Eigen::AngleAxisd(-step, Eigen::Vector3d::UnitZ()));
+    auto target_pose = getPose(path[i], Eigen::Quaterniond(ee_pose.rotation()));
+    target_pose.header.stamp = node->now();
+    pose_publisher->publish(target_pose);
+    rate.sleep();
+    door.rotateDoor(door_angle);
+    door_angle += step;
+    i++;
+  }
```

Comment:
I think range-based for loop will work here since you don't need the index

```suggestion
  for (const auto& pose : path)
  {
    ee_pose.rotate(Eigen::AngleAxisd(-step, Eigen::Vector3d::UnitZ()));
    auto target_pose = getPose(pose, Eigen::Quaterniond(ee_pose.rotation()));
    target_pose.header.stamp = node->now();
    pose_publisher->publish(target_pose);
    rate.sleep();
    door.rotateDoor(door_angle);
    door_angle += step;
  }